src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -54,6 +55,19 @@ public static class PackageExtractor
                 var packageDirectoryInfo = Directory.CreateDirectory(installPath);
                 var packageDirectory = packageDirectoryInfo.FullName;
 
+                if (packageExtractionV2Context.SignedPackageVerifier != null)
```

Comment:
This block of code is duplicated several times throughout this file.  How about extracting it into a helper method?